File: PTT-GC-API/Models/CommonData/CommonData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PTT_GC_API.Models.CommonData
{
    public class CommonData
    {
        [Key]
        public int Id { get; set; }
        [StringLength(255)]
        public string DataType { get; set; }
        [StringLength(1024)]
        public string Attribute01 { get; set; }
        [StringLength(1024)]
        public string Attribute02 { get; set; }
        [StringLength(1024)]
        public string Attribute03 { get; set; }
        [StringLength(1024)]
        public string Attribute04 { get; set; }
        [StringLength(1024)]
        public string Attribute05 { get; set; }
        [StringLength(1024)]
        public string Attribute06 { get; set; }
        [StringLength(1024)]
        public string Attribute07 { get; set; }
        [StringLength(1024)]
        public string Attribute08 { get; set; }
        [StringLength(1024)]
        public string Attribute09 { get; set; }
        [StringLength(1024)]
        public string Attribute10 { get; set; }
        [StringLength(1024)]
        public string Attribute11 { get; set; }
        [StringLength(1024)]
        public string Attribute12 { get; set; }
        [StringLength(1024)]
        public string Attribute13 { get; set; }
        [StringLength(1024)]
        public string Attribute14 { get; set; }
        [StringLength(1024)]
        public string Attribute15 { get; set; }
        [StringLength(1024)]
        public string Attribute16 { get; set; }
        [StringLength(1024)]
        public string Attribute17 { get; set; }
        [StringLength(1024)]
        public string Attribute18 { get; set; }
        [StringLength(1024)]
        public string Attribute19 { get; set; }
        [StringLength(1024)]
        public string Attribute20 { get; set; }

        public bool? IsDelete { get; set; }
        public bool? IsDisable { get; set; }
    }
}

File: PTT-GC-SPA/src/app/core/models/commonData.ts
export interface commonData {
  id: number;
  dataType: string;
  attribute01: string;
  attribute02: string;
  attribute03: string;
  attribute04: string;
  attribute05: string;
  attribute06: string;
  attribute07: string;
  attribute08: string;
  attribute09: string;
  attribute10: string;
  attribute11: string;
  attribute12: string;
  attribute13: string;
  attribute14: string;
  attribute15: string;
  attribute16: string;
  attribute17: string;
  attribute18: string;
  attribute19: string;
  attribute20: string;
}


export interface Currency {
  currencyName: string;
  currencyMillionName: string;
  currencyValue: string;
  fxRate: number;

}

export interface StdProjectDefParams {
  initiativeId: number;
  typeOfInvestment: string;
  budgetSource: string;
}

export interface StdProjectDefRes {
  name: string;
  value: string;
  typeOfInvestment: string;
  budgetSource: string;
  defaultValue: number;
}
